pyproject.toml

```diff
@@ -72,6 +72,10 @@ gcsfs = { version = ">=2023.1.0,<2024.1.0", optional = true }
 psycopg2-binary = { version = ">=2.9.6", optional = true }
 sqlalchemy = { version = "^2.0.18", optional = true }
 getdaft = { version = ">=0.2.12", optional = true }
+botocore = "1.34.51"
+typing-extensions = "4.9.0"
+pydantic-core = "^2.16.3"
+numpy = "1.24"
```

Comment:
Do we want to make these dependencies required? You don't need botocore if you're a GCP user. Also, numpy should be optional if you use s3fs.